src/binary/port.rs

```diff
@@ -273,8 +256,18 @@ pub struct Port<'a, B> {
     /// certainly cause the program to abort rather than unwind the stack) it
     /// can poison the port.
     poison: Option<io::Error>,
-    /// Optional hook to call after a packet is sent/received.
-    packet_hook: Option<PacketCallbackDebugWrapper<'a>>,
+    /// User specified callbacks.
+    callbacks: C,
+    /// Lifetime marker
+    lifetime: std::marker::PhantomData<&'a std::convert::Infallible>,
+}
+
+impl<'a, B: std::fmt::Debug, C> std::fmt::Debug for Port<'a, B, C> {
```

Comment:
Same comment as for `ascii`: probably best to use `Backend` bound instead of `Debug`.